### PR TITLE
Feature/softwareworkshop

### PR DESCRIPTION
**Pull request by pslocum**

### CODE REVIEW

File: Source/Generators/LMCKassSignalGenerator.cc [hunk 1]
```diff
@@ -27,13 +27,14 @@ namespace locust
...
 		fTruth( 0 ),
-        phi_t1(0.),
-        phi_t2(0.),
-        phiLO_t(0.),
-        EventStartTime(-99.),
-        EventToFile(0)
+        fPhi_t1(0.),
+        fPhi_t2(0.),
+        fPhiLO_t(0.),
+		fNPreEventSamples( 1500000 ),
```
Comment (buzinsky) [inline Source/Generators/LMCKassSignalGenerator.cc]: can you fix indenting?

**Comment by buzinsky** [comment]
I just want to make sure (because I didnt know what most of the //pls 's) are about. Did you just remove the pls comments, or were they really things that we think are fine in the current code, that were not ok/ sketchy when it was written. I am ok with removing them in the LMCGlobals, if that is class going to disappear anyways.

Could you give me an example of a case where the previous pls comment was resolved/ made obsolete (ie what was problem before, how was it resolved)

**Comment by pslocum** [comment]
> I just want to make sure (because I didnt know what most of the //pls 's) are about. Did you just remove the pls comments, or were they really things that we think are fine in the current code, that were not ok/ sketchy when it was written. I am ok with removing them in the LMCGlobals, if that is class going to disappear anyways.
> 
> Could you give me an example of a case where the previous pls comment was resolved/ made obsolete (ie what was problem before, how was it resolved)


Here are the explanations for the changes shown below.  

1.  Commented out for compatibility with compiler.
2.  Bug fix.
3.  Hard coded fState to select time domain instead of frequency for compatibility with Kassiopeia. 
4.->10.  Converted (fftw) data type to (int) during upgrade of Locust for digitization of complex data.
11.->14.  Bug fix.

> 

1.  Source/Core/LMCException.hh://#include <Exception>   // pls edit

2.  Source/Core/LMCGeneratorToolbox.cc:                lastGenerator = lastGenerator->GetNextGenerator();  // pls addition to advance list pointer.

3.  Source/Core/LMCSignal.cc:        fState = kTime;  // pls confused here.

4.  Source/Core/LMCSimulationController.cc:                    LWARN( lmclog, "channel " << ch << ": " << index << ": I  " << simulatedSignal->SignalTimeComplex()[ch*recordSize + index][0] << "  " << (int)simulatedSignal->SignalDigitalS()[2*ch*recordSize + index*2] );  // pls added (int)

5.  Source/Core/LMCSimulationController.cc:                    LWARN( lmclog, "channel " << ch << ": " << index << ": Q " << simulatedSignal->SignalTimeComplex()[ch*recordSize + index][1] << "  " << (int)simulatedSignal->SignalDigitalS()[2*ch*recordSize + index*2+1] );  // pls added (int)

6.  Source/Core/LMCSimulationController.cc:                    LWARN( lmclog, "channel " << ch << ": " << index << ": I " << simulatedSignal->SignalTimeComplex()[ch*recordSize + index][0] << "  " << (int)simulatedSignal->SignalDigitalUS()[2*ch*recordSize + index*2] );  // pls added (int)

7.  Source/Core/LMCSimulationController.cc:                    LWARN( lmclog, "channel " << ch << ": " << index << ": Q " << simulatedSignal->SignalTimeComplex()[ch*recordSize + index][1] << "  " << (int)simulatedSignal->SignalDigitalUS()[2*ch*recordSize + index*2+1] );  // pls added (int)

8.  Source/Generators/LMCDigitizer.cc:                    LWARN( lmclog, "digitizing channel " << ch << ": " << index << " I: " << aSignal->SignalTimeComplex()[ch*signalSize + index ][0] << " --> " << (int) digitizedData[2*ch*signalSize + index*2 ] );  // pls added (int)

9.  Source/Generators/LMCDigitizer.cc:                    LWARN( lmclog, "digitizing channel " << ch << ": " << index << " Q: " << aSignal->SignalTimeComplex()[ch*signalSize + index ][1] << " --> " << (int) digitizedData[2*ch*signalSize + index*2+1 ] );  // pls added (int)

10.  Source/Generators/LMCDigitizer.cc:                        LWARN( lmclog, "digitizing channel " << ch << ": " << index << " I: " << aSignal->SignalTimeComplex()[ch*signalSize + index ][0] << " --> " << (int) digitizedData[2*ch*signalSize + index*2 ] );  // pls added (int)
Source/Generators/LMCDigitizer.cc:                        LWARN( lmclog, "digitizing channel " << ch << ": " << index << " Q: " << aSignal->SignalTimeComplex()[ch*signalSize + index ][1] << " --> " << (int) digitizedData[2*ch*signalSize + index*2+1 ] );  // pls added (int)

11.  Source/Generators/LMCDipoleSignalGenerator.cc:        fRequiredSignalState = aDomain;  // pls changed == to =.
12.  Source/Generators/LMCFakeTrackSignalGenerator.cc:        fRequiredSignalState = aDomain;  // pls changed == to =.
13.  Source/Generators/LMCGaussianNoiseGenerator.cc:        fRequiredSignalState = aDomain;  // pls changed == to =.
14.  Source/Generators/LMCTestSignalGenerator.cc:        fRequiredSignalState = aDomain;  // pls changed == to =.

**Comment by buzinsky** [comment]
One more question: in LMCSimulation.cc what does the hard-coded 20 signify?
Thanks for the writeup

**Comment by pslocum** [comment]
The
>
>                 for( unsigned index = 0; index < 20; ++index )
>
with a limit of "20" tells Locust how many digitized voltages to print to
the terminal on completion of the simulation.   It is chosen to avoid
having Locust try to print a large time series to the terminal.

**Comment by buzinsky** [comment]
Looks good. Thanks for going through all of these, I would not have been able

You can merge/ delete the branch at will, if you aren't adding anything else

**Pull request merged**